splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/TriggerHandler.java

```diff
@@ -62,36 +75,174 @@
+        this.txn = txn;
+        this.token = token;
+        Properties properties = new Properties();
+        if (hasStatementTriggerWithReferencingClause) {
+            // Use the smaller of ControlExecutionRowLimit or 1000000 to determine when to switch to spark execution.
+            // Hard cap at 1 million despite the setting of controlExecutionRowLimit since we don't want to exhaust
+            // memory if the sysadmin cranked this setting really high.
+            int switchToSparkThreshold = EngineDriver.driver().getConfiguration().getControlExecutionRowLimit() <= Integer.MAX_VALUE ?
+            (int) EngineDriver.driver().getConfiguration().getControlExecutionRowLimit() : Integer.MAX_VALUE;
+
+            if (switchToSparkThreshold < 0)
+                switchToSparkThreshold = 0;
+            else if (switchToSparkThreshold > 1000000)
+                switchToSparkThreshold = 1000000;
+
+
+            long doubleDetermineSparkRowThreshold = 2 * getLcc().getOptimizerFactory().getDetermineSparkRowThreshold();
+            int inMemoryLimit = switchToSparkThreshold;
+
+            // Pick the larger of switchToSparkThreshold or 2*determineSparkRowThreshold as the
+            // threshold for creating a conglomerate.  By design this will cause the trigger rows
+            // to always be held in memory when executing on control, for performance.
+            // But the interface also supports values of switchToSparkThreshold which are larger than
+            // overflowToConglomThreshold, in which case we could create a temporary conglomerate while
+            // executing in control.
+            int overflowToConglomThreshold =
+                 doubleDetermineSparkRowThreshold > inMemoryLimit ? (int)doubleDetermineSparkRowThreshold :
+                 doubleDetermineSparkRowThreshold < 0 ? 0 :
+                 inMemoryLimit;
+
+            // Spark doesn't support use of an in-memory TriggerRowHolderImpl, so we
+            // set overflowToConglomThreshold to zero and always use a conglomerate.
+            if (isSpark) {
+                switchToSparkThreshold = 0;
+                overflowToConglomThreshold = 0;
+            }
+
+            triggerRowHolder =
+                new TriggerRowHolderImpl(activation, properties, writeInfo.getResultDescription(),
+                                         overflowToConglomThreshold, switchToSparkThreshold,
+                                         templateRow, tableVersion, isSpark, txn, token, ConglomID,
+                                          this.getTriggerExecutionContext());
+        }
+
+    }
+
+    public TriggerExecutionContext getTriggerExecutionContext() { return this.triggerActivator.getTriggerExecutionContext(); }
+
     private void initTriggerActivator(Activation activation, WriteCursorConstantOperation constantAction) throws StandardException {
         try {
+            withContext = new Function<Function<LanguageConnectionContext,Void>, Callable>() {
```

Comment:
`            withContext = f -> (Callable) () -> {
                ActivationHolder ah = new ActivationHolder(activation, null);
                try {
                ...
`

Reply:
Is that shorthand or something?  Is there anything wrong with the code in its current form?  I'm just porting from Daniel's branch.